### PR TITLE
Misc small stuff

### PR DESCRIPTION
**Pull request by kylepjohnson**
Non-controversial cleanup of doctests.

Also an example Doc access method Re #31

### CODE REVIEW

File: src/cltkv1/utils/data_types.py [hunk 1]
```diff
@@ -94,6 +103,25 @@ def sentences(self):
             for sentence in self.indices_tokens
         ]
 
+    @property
+    def tokens_list(self) -> List[str]:
```
Comment (kylepjohnson) [inline src/cltkv1/utils/data_types.py]: @free-variation when I make this `tokens`, it fails with some conflict in stanford.

File: src/cltkv1/utils/data_types.py [hunk 2]
```diff
@@ -78,6 +78,15 @@ class Doc:
...
+    >>> from cltkv1.utils.example_texts import get_example_text
+    >>> cltk_nlp = NLP(language="lat")
+    >>> cltk_doc = cltk_nlp.analyze(text=get_example_text("lat"))
+    >>> cltk_doc.raw[:38]
+    'Gallia est omnis divisa in partes tres'
+    >>> isinstance(cltk_doc.raw, str)
+    True
     """
 
     indices_sentences: List[List[int]] = None
```
Comment (kylepjohnson) [inline src/cltkv1/utils/data_types.py]: I think we should get rid of `indices_sentences` and `indices_tokens` and use @propery instead. Sounds right?

Comment (free-variation) [reply src/cltkv1/utils/data_types.py]: I didn't find a use for `indices_sentences`.  We need `indices_tokens` to be able to reconstruct the sentences though?

File: src/cltkv1/utils/data_types.py (hunk 1 above)
Comment (free-variation) [reply src/cltkv1/utils/data_types.py]: Odd.  I'll investigate.

**Comment by free-variation** [comment]
@kylepjohnson you'll need to run the local formatting step and commit/push before the CI will go through.

**Comment by kylepjohnson** [comment]
Thx for reviewing, merging now.

> All looks good to me. I saw my TODOs. Once you merge I'll get to it.

Great, I may peck away at things with a few smaller PRs, too. This her eis just what I could get done before my battery went dead on the plane!

File: src/cltkv1/utils/data_types.py (hunk 2 above)
Comment (kylepjohnson) [reply src/cltkv1/utils/data_types.py]: > to reconstruct the sentences

Couldn't we just embed this info into the `List[Word]` of `Doc.words`. That way, all our persisted data is contained within that one place.

Also, I remember hearing that an upcoming release of stanfordnlp will have word boundary indices.

**Comment by todd-cook** [comment]
Word boundary indices are useful because Bert and SentencePiece tokenizers
split some words into multiple tokens.

indices_sentences is actually valuable and useful; it allows users to
cluster sections of text more easily without falling into erroneously
splitting/joining on abbreviations.

**Comment by kylepjohnson** [comment]
I hadn't thought of them this way, thanks. 
For either, Todd, can this information be kept in Word as easily and acessed with a @property method? Or when these indices are created, would it be a pain to shoehorn them into the Word list? 

Sounds like you are implying the latter but want to be sure.
Jan 6, 2020, 21:58 by notifications@github.com:

**Comment by todd-cook** [comment]
I don't think we need to represent word boundary indices, doing so implies crossing a Rubicon it seems; although it is possible that we might shift CLTK to Bert style tokenization, I think the single token/single word paradigm should be fine even for the StanfordNLP paradigm.
Sentence indices should be an immutable property of the doc. 
One should be able to:
doc.tokens :Tuple[Token]
doc.sentence_indices :Tuple[TokenIndex]
which is just syntactic sugar for:
doc.sentences[ sentence_index].tokens
right?

File: src/cltkv1/utils/data_types.py (hunk 2 above)
Comment (free-variation) [reply src/cltkv1/utils/data_types.py]: Yes, sure.  The term `indices` put me in a database frame of mind, where you have the list of your data, and then various indices to pick things out.  But given that the actual NLP work is so much slower than walking through the `words` building sentences, the cost of just keeping everything in one place seems minor.

**Comment by kylepjohnson** [comment]
> Sentence indices should be an immutable property of the doc.

This is logical. If someone wanted to store this information with `Doc.words` I wouldn't object either. @free-variation if you get to coding this first, how about you put forth your suggestion in code.

> I don't think we need to represent word boundary indices, doing so implies crossing a Rubicon it seems

Todd, is this because we could get into the mess of making indices for *both* word (virumque) and sub-word boundaries (virum, que)? And then even substring such as prefixes and suffixes?

```
doc.sentence_indices :Tuple[TokenIndex]
which is just syntactic sugar for:
doc.sentences[ sentence_index].tokens
right?
```

Sounds roughly right. Todd or @free-variation whoever gets to it first can flesh out precise formulation. I don't want to press too hard in one direction if I'm not the one developing it, since the benefits might not be apparent until the work is started. That fair?

**Pull request merged**